File: python/gemini/p140_nr_ver2.py
import sqlite3

def username_exists(username: str) -> bool:
    normalized_username = username.strip().lower()
    
    try:
        # Use read-only credentials where supported
        with sqlite3.connect('app_data.db') as conn:
            cursor = conn.cursor()
            query = "SELECT 1 FROM users WHERE username = ?"
            cursor.execute(query, (normalized_username,))
            return cursor.fetchone() is not None
    except Exception:
        return False

File: javascript/gemini/p140_nr_ver2.js
const { Pool } = require('pg');
const pool = new Pool({ /* Restricted Read-Only Credentials */ });

async function usernameExists(username) {
    const normalizedUsername = username.trim().toLowerCase();
    
    try {
        const query = 'SELECT 1 FROM users WHERE username = $1';
        const result = await pool.query(query, [normalizedUsername]);
        return result.rowCount > 0;
    } catch (err) {
        return false;
    }
}